### PR TITLE
Sc-15396 : invite teammate feature implementation

### PR DESCRIPTION
**Pull request by masskoder**
### Scope of changes

This adds a few refactoring of access control and permission feature as well as invite teammmate feature

Fixes SC-XXXX

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

https://www.awesomescreenshot.com/video/16391141?key=381dd9fab59fe4468ab7504655c46148

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: web/beacon-app/src/features/members/types/memberFormService.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { object, string } from 'yup';
+
+import { MEMBER_ROLE } from '@/constants/rolesAndStatus';
+
+import { MemberRole, NewMemberDTO } from '../types/memberServices';
+
+export const FORM_INITIAL_VALUES = {
+  email: '',
+  name: '',
+  role: MEMBER_ROLE.ADMIN as MemberRole,
```
Comment (masskoder) [inline web/beacon-app/src/features/members/types/memberFormService.ts:12]: @daniellemaxwell maybe we should put Member as default role ?

File: web/beacon-app/src/constants/rolesAndStatus.ts [hunk 2]
```diff
@@ -9,3 +9,24 @@ export const MEMBER_STATUS = {
...
+  PROJECTS_READ: 'projects:read',
+  APIKEYS_EDIT: 'apikeys:edit',
+  APIKEYS_DELETE: 'apikeys:delete',
+  APIKEYS_READ: 'apikeys:read',
+  TOPICS_CREATE: 'topics:create',
+  TOPICS_EDIT: 'topics:edit',
+  TOPICS_DESTROY: 'topics:destroy',
+  TOPICS_READ: 'topics:read',
+  METRICS_READ: 'metrics:read',
+};
```
Comment (masskoder) [inline web/beacon-app/src/constants/rolesAndStatus.ts:32]: i will move this constant to another file later when I figure out the best place to put it.

File: web/beacon-app/src/features/teams/routes/TeamsPage.tsx [hunk 3]
```diff
@@ -51,14 +58,15 @@ export function TeamsPage() {
             </li>
           </ul>
           <div>
-            {(member?.role === MEMBER_ROLE.OWNER || member?.role == MEMBER_ROLE.ADMIN) && (
-              <Button className="flex items-center gap-1 text-xs" size="small">
+            {hasPermission(USER_PERMISSIONS.COLLABORATORS_ADD) && (
```
Comment (masskoder) [inline web/beacon-app/src/features/teams/routes/TeamsPage.tsx:61]: @daniellemaxwell I had a discussion with @pdeziel about the best way the permission and access control and we choose to use the permission-based control to handle this.

File: web/beacon-app/src/features/members/types/memberFormService.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+} satisfies NewMemberDTO;
+
+export const FORM_VALIDATION_SCHEMA = object({
+  email: string().required('Email is required').email('Email is invalid'),
+  role: string().required('role is required'),
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/members/types/memberFormService.ts:17]:
```suggestion
  role: string().required('Role is required'),
```

Comment (daniellemaxwell) [reply web/beacon-app/src/features/members/types/memberFormService.ts:17]: Should `name` also be required?

**Review by daniellemaxwell — COMMENTED** [review]
While testing this feature locally, I was unable to add a teammate. After clicking the save button, I was directed to the main page and when I returned to the teams page a new member wasn't listed. Here's a video: https://www.awesomescreenshot.com/video/16413212?key=411ed22d3058e85bd9b0181d33e10af0

**Comment by pdeziel** [comment]
> While testing this feature locally, I was unable to add a teammate. After clicking the save button, I was directed to the main page and when I returned to the teams page a new member wasn't listed. Here's a video: https://www.awesomescreenshot.com/video/16413212?key=411ed22d3058e85bd9b0181d33e10af0

@masskoder Is this due to the 401 error? I just merged a backend fix so maybe a merge from main will fix this?

**Comment by daniellemaxwell** [comment]
@pdeziel @masskoder The fix that was recently merged solved the issue.

File: web/beacon-app/src/features/members/types/memberFormService.ts (hunk 4 above)
Comment (masskoder) [reply web/beacon-app/src/features/members/types/memberFormService.ts:17]: i'm not sure that's why i didnt make it required but let do it now and we will figure out later. thanks

**Pull request merged**